x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -1401,3 +1401,38 @@ export async function incrementPackageName(
 
   return getMaxPackageName(packageName, packagePolicyData?.items);
 }
+
+export async function incrementPackagePolicyCopyName(
```

Comment:
There is a similar function to increment package policy name, you could make it more generic to work with copy use case as well: https://github.com/elastic/kibana/blob/dec594b303061787f31cbf6170bf30a363ecf14b/x-pack/plugins/fleet/server/services/package_policy.ts#L1392-L1402

Reply:
They are very similar but abstracting the logic and make it generic will probably make it less readable and increase the complexity. In this case I think it make sense to have two distinct simple functions.

Reply:
I still don't like this duplication, the logic of finding the maximum number in package policy names is the same. Maybe you could pass the regex as a parameter?

Also, package_policy service is already very long (1400+ lines) so we shouldn't be adding utility functions here.

Reply:
> I still don't like this duplication, the logic of finding the maximum number in package policy names is the same. Maybe you could pass the regex as a parameter?

I think it's okay to have very similar code and sometimes is better than over abstracting, in this case it make things more readable and easier to maintain. If we had an other function similar to that one it will make sense to revisit that.

> Also, package_policy service is already very long (1400+ lines) so we shouldn't be adding utility functions here.

We could probably move the utility helper to a new file, I am going to take a shoot.

Reply:
Just added the new helper file where I moved the two `increment*` function